src/hangar/dataloaders/torch_loader.py

```diff
@@ -0,0 +1,97 @@
+from ..dataset import DatasetDataReader
+try:
+    import torch
+    from torch.utils.data import Dataset, DataLoader
+except (ImportError, ModuleNotFoundError):
+    raise RuntimeError("Could not import torch. Try installing extra dependencies")
+
+
+class TorchDataSet(Dataset):
+    """
+    TorchDataSet inherits `torch.utils.data.Dataset` and override
+    `__len__` & `__getitem__`. Not needed to use directly by the user.
+
+    Parameters
+    ----------
+    hangar_datasets : `hangar.dataset.DatasetDataReader` or a Collection
+        A dataset object or a list of dataset objects which will be
+        consumed by :class:`TorchDataSet`
+    """
+
+    def __init__(self, hangar_datasets):
+        if isinstance(hangar_datasets, DatasetDataReader):
+            hangar_datasets = [hangar_datasets]
+
+        try:
+            self._dataset_len = len(hangar_datasets[0])
+        except IndexError:
+            raise
+        except Exception:
+            raise TypeError("`hangar_datasets` has to be a valid hangar dataset or an"
+                            "iterable that can hold other objects, like a list.")
+
+        for dset in hangar_datasets:
+            if not isinstance(dset, DatasetDataReader):
+                raise TypeError("`hangar_dataset` contains invalid hangar dataset(s)")
+            if len(dset) != self._dataset_len:
+                raise RuntimeError("Got datasets with different length")
+        self._number_of_datasets = len(hangar_datasets)
+        self.hangar_datasets = hangar_datasets
+        self._data_names = tuple(self.hangar_datasets[0].keys())
+
+    def __len__(self):
+        return self._dataset_len
+
+    def __getitem__(self, index):
+        key = self._data_names[index]
+        if self._number_of_datasets == 1:
+            return torch.from_numpy(self.hangar_datasets[0][key])
```

Comment:
I would like the return type of the function to be consistent. Having to re-wrap items in a list for consistent downstream handling is annoying.

Also, in 1.1 if a dataset returns a named tuple, the data loader will preserve that tuple type as the return value. I'd like `__init__` to pick up a `tuple_cls=tuple` parameter, and have this function `return self._tuple_cls(out)`.

Reply:
`_tuple_cls` sounds brilliant and never knew about named tuple thing. What do you think about making 1.1 the lowest version we support?

Reply:
Never knew new named tuple thing, thanks for the pointer. So `tuple_cls` seemed brilliant but that actually creates more problems. For example, `tuple` expect a single argument which can be a list/tuple while named tuple expect n number of arguments where n is the number of fields. We'll have to do another check for that which I think is weirder. 
Also, the rationale behind returning non-consistent return type is although it is inconsistent in different cases, it is consistent with the input type. If the input type is single `hangar.Dataset` output is a single batch. If the input is a list of `hangar.Dataset` output is a list of batches. Also, regardless of tuple/list torch DataSet returns, Dataloader always returns a list.
So I think we can accept two types of inputs, 
1. `hangar.Dataset`
2. `list` / `tuple` of `hangar.Dataset`

and if `tuple_cls` (default to `None`) is defined then without doing any check we'll return `tuple_cls(out)` otherwise return the output as it is

What do you think?

Reply:
If I can get a one-element tuple in the one-dataset case by setting `tuple_cls` to `tuple` then I'm happy.

You can check for `hasattr(self._tuple_cls, '_fields')` to detect the named tuple case.  :)

Reply:
That's the thing, PyTorch DataLoaders convert that tuple to a one-element list. So you break the consistency of return data type there itself.

Reply:
Dataset now returns a list, tuple or a namedtuple. Nothing else

Reply:
I am thinking of returning only namedtuple with hangar dataset names as field names (if the user doesn't pass a `tuple_cls`) What do you think @elistevens

Reply:
Oh, I like that idea a lot.